snapcraft/cli/store.py

```diff
@@ -336,14 +337,18 @@ def export_login(login_file: TextIO, snaps: str, channels: str, acls: str,
                            save=False):
         sys.exit(1)
 
-    store.conf.save(config_fd=login_file)
+    with open(login_file, 'w') as f:
+        store.conf.save(config_fd=f)
+
+    # This is sensitive-- make sure it's only readable by the owner
+    os.chmod(login_file, stat.S_IRUSR)
```

Comment:
there's some time  here though in which the mode is still permissive,  in similar code I ended up doing something like this (there are probably other ways):

    private_open = functools.partial(os.open, mode=0o600)
    with open(fn, 'w', opener=private_open) as f:
       ...

Reply:
Ah, good call, thanks @pedronis. Think I should leave it at 0o600, then, or go all the way to 0o400 after it's been written?

Reply:
Fixed-- I still ratchet all the way down to 0o400 unless someone says otherwise.